File: core-network/src/main/java/com/levnovikov/core_network/HttpClientFactory.java
package com.levnovikov.core_network;

import com.levnovikov.core_network.transformers.AuthTransformer;

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

public class HttpClientFactory {

    private final AuthTransformer authTransformer;

    HttpClientFactory(AuthTransformer authTransformer) {
        this.authTransformer = authTransformer;
    }

    public HttpClient buildNetworkClient() {
        return new HttpClientImpl.Builder()
//                .threadComposer(new NetworkThreadComposer()) TODO solve
                .callExecutor(new UrlConnectionCallExecutor())
                .addTransformer(authTransformer)
                .build();
    }

    public HttpClient buildImageLoaderClient() {
        return new HttpClientImpl.Builder()
//                .threadComposer(new ImageLoaderThreadComposer()) TODO solve
                .callExecutor(new UrlConnectionCallExecutor())
                .build();
    }
}

File: core-network/src/main/java/com/levnovikov/core_network/HttpClientFactory.kt
package com.levnovikov.core_network

import com.levnovikov.core_network.transformers.AuthTransformer

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

class HttpClientFactory internal constructor(private val authTransformer: AuthTransformer) {

    fun buildNetworkClient(): HttpClient {
        return HttpClientImpl.Builder()
                .callExecutor(UrlConnectionCallExecutor())
                .addTransformer(authTransformer)
                .build()
    }

    fun buildImageLoaderClient(): HttpClient {
        return HttpClientImpl.Builder()
                .callExecutor(UrlConnectionCallExecutor())
                .build()
    }
}
